server/fileCompress/app.py

```diff
@@ -0,0 +1,103 @@
+import os
+import zipfile
+import json
+import posixpath
+from pprint import pprint
+import boto3
+import shutil
+
+
+def lambda_handler(event, context):
+    # Define the constants
+    TABLE_NAME = os.environ["JOB_TABLE"]
+    BUCKET_NAME = os.environ["SRC_BUCKET"]
+    PRIMARY_KEY = "id"
+    LOOKUP_BATCH_ID = event["queryStringParameters"]["id"]
+
+    # Create table and bucket reference
+    table = boto3.resource("dynamodb").Table(TABLE_NAME)
+    bucket = boto3.resource("s3").Bucket(BUCKET_NAME)
+
+    # Get the data from DynamoDB
+    response = table.get_item(Key={PRIMARY_KEY: LOOKUP_BATCH_ID})
+
+    # Download files using reference data from DynamoDB
+    files_json = response["Item"]["files"]
+
+    # Create zip file
+    zip_file_path = os.path.join("/tmp/", "temp_zip.zip")
+    temp_zip = zipfile.ZipFile(zip_file_path, "w")
+
+    # Go through each file
+    for file in files_json:
+
+        # Get file path for s3 bucket and create file path to store in tmp
+        file_ref_s3 = file["fileRef"]
+        file_ref_tmp = "/tmp/" + file["fileRef"].split("/")[-1]
```

Comment:
Inconsistent with previous path creation. You should be using Path(file["fileRef"]).parts[-1] to get the last part.

Reply:
Again, not sure what you mean by Path. I did os.path.split(file["fileRef"])[-1] which I believe is the same functionality